service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/UploadCMOServiceTest.java

```diff
@@ -0,0 +1,427 @@
+        Element<HearingBooking> hearing = element(
+            hearing(CASE_MANAGEMENT, LocalDateTime.of(2020, 1, 15, 0, 0), cmo.getId())
+        );
+
+        hearings.add(hearing);
+
+        Map<String, Object> initialPageData = service.getInitialPageData(hearings, List.of(cmo));
+
+        Map<String, Object> expected = Map.of(
+            "pastHearingList", dynamicList(hearings.get(0).getId(), hearings.get(1).getId(), hearings.get(2).getId()),
+            "numHearings", "MULTI",
+            "multiHearingsWithCMOs", "Case management hearing, 15 January 2020",
+            "showHearingsMultiTextArea", "YES"
+        );
+
+        assertThat(initialPageData).isEqualTo(expected);
+    }
+
+    @Test
+    void shouldReturnCorrectPageDataWhenThereIsOneRemainingHearing() {
+        List<Element<HearingBooking>> hearings = List.of(element(
+            hearing(CASE_MANAGEMENT, LocalDateTime.of(2020, 2, 1, 0, 0))
+        ));
+
+        Map<String, Object> initialPageData = service.getInitialPageData(hearings, List.of());
+
+        Map<String, Object> expected = Map.of(
+            "numHearings", "SINGLE",
+            "cmoHearingInfo", "Send agreed CMO for Case management hearing, 1 February 2020."
+                + "\nThis must have been discussed by all parties at the hearing.",
+            "cmoJudgeInfo", "His Honour Judge Dredd"
+        );
+
+        assertThat(initialPageData).isEqualTo(expected);
+    }
+
+    @Test
+    void shouldReturnCorrectPageDataWhenThereIsOneRemainingHearingWithSomeAlreadyAssigned() {
+        Element<CaseManagementOrder> cmo = element(CaseManagementOrder.builder().build());
+        List<Element<HearingBooking>> hearings = List.of(
+            element(hearing(CASE_MANAGEMENT, LocalDateTime.of(2020, 2, 1, 0, 0))),
+            element(hearing(CASE_MANAGEMENT, LocalDateTime.of(2020, 2, 2, 0, 0), cmo.getId()))
+        );
+
+        Map<String, Object> initialPageData = service.getInitialPageData(hearings, List.of(cmo));
+
+        Map<String, Object> expected = Map.of(
+            "numHearings", "SINGLE",
+            "cmoHearingInfo", "Send agreed CMO for Case management hearing, 1 February 2020."
+                + "\nThis must have been discussed by all parties at the hearing.",
+            "cmoJudgeInfo", "His Honour Judge Dredd",
+            "singleHearingsWithCMOs", "Case management hearing, 2 February 2020",
+            "showHearingsSingleTextArea", "YES"
+        );
+
+        assertThat(initialPageData).isEqualTo(expected);
+    }
+
+    @Test
+    void shouldReturnCorrectPageDataWhenThereAreNoRemainingHearings() {
+        Map<String, Object> initialPageData = service.getInitialPageData(List.of(), List.of());
+
+        Map<String, String> expected = Map.of(
+            "numHearings", "NONE"
+        );
+
+        assertThat(initialPageData).isEqualTo(expected);
+    }
+
+    @Test
+    void shouldReturnCorrectPageDataWhenThereAreNoRemainingHearingsWithoutCMOsAssigned() {
```

Comment:
I am not sure if I like "shouldReturnCorrectPageData" of course the code should work. It's quite wordy but a bit obvious, would you consider rename?